src/librustc_typeck/check/op.rs

```diff
@@ -707,11 +715,15 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                                 hir::UnOp::UnNot => "std::ops::Not",
                                 hir::UnOp::UnDeref => "std::ops::UnDerf",
                             };
-                            err.note(&format!(
-                                "an implementation of `{}` might \
-                                                be missing for `{}`",
-                                missing_trait, operand_ty
-                            ));
+                            if let Adt(def, _) = operand_ty.peel_refs().kind {
```

Comment:
Given that we're doing this three times, can we make a closure or function and call that three times instead?